CONTRIBUTING.md

```diff
@@ -0,0 +1,269 @@
+- **Melos**: Serverpod uses Melos to manage the monorepo. You can install Melos by running `pub global activate melos`.
+- **bash**: Some scripts require bash to be installed on your machine. If you are on Windows, you can install Git Bash from the [Git website](https://git-scm.com/downloads).
+
+After the required tools have been installed, you will need a local clone of the repository.
+We recommend [forking](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo) the repository and then cloning your fork.
+
+After the repository is cloned, run the following command to install all dependencies:
+
+```bash
+$ util/pub_get_all
+```
+
+> [!TIP]
+> If you have recently configured a Serverpod project, you can add the `--offline` flag to the script above to use cached versions of the dependencies.
+
+The project is now set up and ready for development.
+
+### Activating the CLI
+
+Activate the `serverpod` command from your cloned repository by running the following commands:
+
+```bash
+$ cd tools/serverpod_cli
+$ dart pub get
+$ dart pub global activate --source path .
+```
+
+Depending on your Dart version you may need to run the `dart pub global` command above every time you've made changes in the Serverpod tooling.
+
+> [!TIP]
+> At the time of writing, a bug in `pub global activate` prevents changes from being picked up in activated libraries, tracked here: https://github.com/dart-lang/pub/issues/4295.
+>
+> If you are experiencing issues with the CLI not picking up changes, you can try running the workarounds listed in the issue.
+
+
+When projects are created using your cloned version of the CLI, all serverpod dependencies in the `pubspec.yaml` files will point to your cloned repo.
+
+#### Using local templates
+
+To use templates from your cloned repository, you will need to set the `SERVERPOD_HOME` environment variable. It should point to the root your cloned `serverpod` monorepo. (E.g. `/Users/myuser/MyRepos/serverpod`)
+
+### Code style
+
+Serverpod uses the Dart linter to enforce a consistent code style. The linter is run as part of the CI checks, so it is important that the code follows the linter rules. When you write code, make sure to use `dart format` and `dart analyze` to ensure that the code follows the linter rules.
+
+We try to follow the [Effective Dart](https://dart.dev/guides/language/effective-dart) guidelines as much as possible. But above all we care about code readability and maintainability. Therefore, we encourage you to write code that is easy to read and understand for future contributors.
+
+### Running the tests
+
+Serverpod has a comprehensive test suite that covers the core functionality of the project. The tests are run as part of the CI checks, so to speed up development it can be good to run the tests locally before submitting a pull request.
```

Comment:
```suggestion
Serverpod has a comprehensive test suite that covers the core functionality of the project. The tests are run as part of the CI checks, but to speed up development it can be good to run the tests locally before submitting a pull request.
```